src/command_modules/azure-cli-policyinsights/azure/cli/command_modules/policyinsights/custom.py

```diff
@@ -0,0 +1,260 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from msrestazure.tools import is_valid_resource_id, resource_id
+from azure.cli.core.commands.client_factory import get_subscription_id
+
+
+def list_policy_events(
+        cmd,
+        client,
+        management_group_name=None,
+        resource_group_name=None,
+        resource=None,
+        resource_provider_namespace=None,
+        resource_type=None,
+        policy_set_definition_name=None,
+        policy_definition_name=None,
+        policy_assignment_name=None,
+        from_value=None,
+        to_value=None,
+        order_by_clause=None,
+        select_clause=None,
+        top_value=None,
+        filter_clause=None,
+        apply_clause=None):
+
+    from azure.mgmt.policyinsights.models import QueryOptions
+
+    query_options = QueryOptions(
+        top=top_value,
+        order_by=order_by_clause,
+        select=select_clause,
+        from_property=from_value,
+        to=to_value,
+        filter=filter_clause,
+        apply=apply_clause)
+
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+
+    if policy_assignment_name is not None:
```

Comment:
Can the names be an empty string? If not, change this to `if policy_assignment_name:`.
Same for others below.

Reply:
Will do